torch/_functorch/_aot_autograd/autograd_cache.py

```diff
@@ -3,16 +3,86 @@
 """
 from __future__ import annotations
 
+import functools
 import logging
+import os
 
 import torch
-from torch._inductor.codecache import _ident, FxGraphCachePickler
+from torch._inductor.codecache import (
+    _ident,
+    FxGraphCachePickler,
+    get_code_hash,
+    get_inductor_root,
+)
+from torch.fx.node import Node
 
 from .schemas import AOTConfig  # noqa: F401
 
 log = logging.getLogger(__name__)
 
 
+class BypassAOTAutogradCache(Exception):
+    pass
+
+
+def check_node_safe(node: Node):
+    """
+    Checks that the node only uses supported operators.
+    """
+
+    def is_torch_function(target):
+        if isinstance(target, str):
+            name = target
+        elif callable(target):
+            name = target.__name__
+        else:
+            return False
+        # TODO: is this the right check?
+        return name.startswith("torch.")
```

Comment:
Chatting with @albanD, he pointed out that most of our torch ops (the ops that have direct implementations bound to python from C) have a `builtin` python type:
```
>>> type(torch.add)
<class 'builtin_function_or_method'>
```

So "does fn name start with `torch` and is it a builtin type" sounds like a safe way to check. This also avoids the monkey-patch problem, since monkeypatching `torch.add` to something custom will change its type.

Although this strategy relies on dynamo inlining most user code directly into the torch.* implementations. If we start with that, we can at least see how much of the dynamo graph this check would cover.

For example: our API's in `torch.nn.functional.*` are python functions that eventually call raw `torch.*` API's; so if dynamo blindly allow-in-graphs `torch.nn.functional.*` API's then they wouldn't fit into this check. And... I don't think dynamo inlines `torch.nn.functional.*` API's today (maybe it should?)